openmls/src/group/mls_group/tests_and_kats/tests/proposals.rs

```diff
@@ -438,147 +395,62 @@ fn test_group_context_extension_proposal_fails(
-        QueuedProposal::from_authenticated_content_by_ref(
-            ciphersuite,
-            provider.crypto(),
-            bob_add_proposal,
-        )
-        .unwrap(),
-    );
+    alice_group.merge_pending_commit(provider).unwrap();
 
-    log::info!(" >>> Creating commit ...");
-    let params = CreateCommitParams::builder()
-        .framing_parameters(framing_parameters)
-        .force_self_update(false)
-        .build();
-    let create_commit_result = alice_group
-        .create_commit(params, provider, &alice_signer)
-        .expect("Error creating commit");
-
-    log::info!(" >>> Staging & merging commit ...");
-
-    alice_group
-        .merge_commit(provider, create_commit_result.staged_commit)
-        .expect("error merging pending commit");
-    let ratchet_tree = alice_group.public_group().export_ratchet_tree();
-
-    let _bob_group = StagedCoreWelcome::new_from_welcome(
-        create_commit_result
-            .welcome_option
-            .expect("An unexpected error occurred."),
-        Some(ratchet_tree.into()),
-        bob_key_package_bundle,
+    let ratchet_tree = alice_group.export_ratchet_tree();
+
+    let _bob_group = StagedWelcome::new_from_welcome(
         provider,
-        ResumptionPskStore::new(1024),
+        &MlsGroupJoinConfig::default(),
+        welcome.into_welcome().unwrap(),
+        Some(ratchet_tree.into()),
     )
-    .and_then(|staged_join| staged_join.into_core_group(provider))
+    .and_then(|staged_join| staged_join.into_group(provider))
     .expect("Error joining group.");
 
     // TODO: openmls/openmls#1130 re-enable
     // // Now Bob wants the ApplicationId extension to be required.
     // // This should fail because Alice doesn't support it.
-    // let e = bob_group
-    //     .create_group_context_ext_proposal(
-    //         framing_parameters,
-    //         &alice_credential_bundle,
-    //         &[required_application_id],
-    //         provider,
-    //     )
-    //     .expect_err("Bob was able to create a gce proposal for an extension not supported by all other parties.");
-    // assert_eq!(
-    //     e,
-    //     CreateGroupContextExtProposalError::TreeSyncError(
-    //         crate::treesync::errors::TreeSyncError::UnsupportedExtension
-    //     )
-    // );
+    //let unsupported_extensions = Extensions::single(Extension::Unknown(
+    //    0xff00,
+    //    UnknownExtension(vec![0, 1, 2, 3]),
+    //));
+    //let e = bob_group
+    //    .propose_group_context_extensions(provider, unsupported_extensions, &bob_signer)
+    //    .expect_err("Bob was able to propose an extension not supported by all other parties.");
+    //
+    //assert_eq!(
+    //    e,
+    //    ProposalError::CreateGroupContextExtProposalError(
+    //        CreateGroupContextExtProposalError::KeyPackageExtensionSupport(
+    //            KeyPackageExtensionSupportError::UnsupportedExtension
+    //        )
+    //    )
+    //);
```

Comment:
Curious what is blocking this, as #1130 is not very explicit. Is it unsupported extensions? It's fine if that's out-of-scope if it's too complicated.

Reply:
I'm not sure, I just updated the code to work with the `MlsGroup` and ran it to see if the issue had resolved itself since the issue was filed (it hasn't).

I think this is about OpenMLS ensuring that everyone in a group actually has the capability to support an extension that the application wants to introduce via this proposal. I think we already have a check that would prevent such a proposal from being committed, but again, I haven't really checked.

I think fixing and properly testing this is better done in a separate PR.